
#pragma once

// Qt
#include <QWidget>

// Application
#include "dommodel.h"

//-------------------------------------------------------------------------------------------------
// Forward declarations

class CMainController;

namespace Ui {
    class CDOMView;
}

//-------------------------------------------------------------------------------------------------

class CDOMView : public QWidget
{
    Q_OBJECT

public:

    //-------------------------------------------------------------------------------------------------
    // Constructors and destructor
    //-------------------------------------------------------------------------------------------------

    //! Constructor
    explicit CDOMView(QWidget *parent=0);

    //-------------------------------------------------------------------------------------------------
    // Control methods
    //-------------------------------------------------------------------------------------------------

    //!
    void openFile(const QString& sFileName);

    //-------------------------------------------------------------------------------------------------
    // Protected control methods
    //-------------------------------------------------------------------------------------------------

protected:

    //!
    virtual void resizeEvent(QResizeEvent* event) Q_DECL_OVERRIDE;

    //!
    void openFileInternal(const QString& sFileName);

    //!
    void setXMLFile(const QString& sFileName);

    //-------------------------------------------------------------------------------------------------
    // Slots
    //-------------------------------------------------------------------------------------------------

protected slots:

    //!
    void onBrowseClicked(bool bValue);

    //-------------------------------------------------------------------------------------------------
    // Properties
    //-------------------------------------------------------------------------------------------------

private:

    Ui::CDOMView*   ui;
    DomModel*       m_mModel;
    QString         m_sXMLPath;
};
